pkg/codegen/nodejs/importer.go

```diff
@@ -69,6 +69,24 @@ type NodePackageInfo struct {
 
 	// Respect the Pkg.Version field in the schema
 	RespectSchemaVersion bool `json:"respectSchemaVersion,omitempty"`
+
+	// Set of optimization strategies for optimizing node module
+	// loading in the generated code. By default the code eagerly
+	// loads all modules.
+	//
+	// Supported strategies include:
+	//
+	//     "lazy-load-functions"
+	//     "lazy-load-resources"
+	//     "use-type-only-enums-references"
+	//
+	// Optimizing module loading is relevant for larger providers
+	// like azure-native.
+	//
+	// Note that "use-type-only-enums-references" is experimental.
+	// It requires TypeScript 3.8 or higher to compile the
+	// generated code, and may in some cases fail to generate compiling code.
+	OptimizeNodeModuleLoading []string `json:"optimizeNodeModuleLoading,omitempty"`
```

Comment:
Feature flags to opt-in to the changes here.

Reply:
Is there a reason someone wouldn't want to opt-in?